packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,702 @@
+
+In addition, to the `defaultColumnName` property, let's add the `defaultFilterValue` property to allow a user to apecify initial filter value. When a user selects a value in our select box, this value will be changed. In this case, save it in the state object of our component as shown below:
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+
+export class ToolbarFilterState extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filterValue: this.props.defaultFilterValue
+    };
+    ...
+  }
+
+  render() {
+    ...
+  }
+}
+```
+In our scenario, we are going to filter the grid by the `CompanyName` column. The default filter value should be set to an empty string which means that the grid should not fitler items initially. So, we can add this plugin to our grid. Since it is a state management plugin it should be included before UI plugins. See [React Grid - Plugin Overview](./plugin-overview.md).
+
+*index.js:*
+```jsx
+...
+import {ToolbarFilterState} from './plugins/toolbar-filter-state.js';
+
+class App extends Component {
+  ...
+  render() {
+    ...
+    return (
+      <Paper>
+        <Grid
+          ...
+        >
+          <ToolbarFilterState defaultColumnName="CompanyName" defaultFilterValue="" />
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+          <ToolbarFilter />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+We added the `ToolbarFilterState` plugin but nothing has changed and the select box still does nothing. The next task is to get unique values of the `CompanyName` column and pass these values to `ToolbarFilter` for displaying. If we want to share data between plugins, we need to use [getters](../../../core/docs/reference/getter.md).
+
+How can we get values of a particular column? We set data items using the `rows` property of the [Grid](../reference/grid.md) plugin. This plugin exports the `rows` getter. We can use it in our case. However, we will need to pass data items to our UI plugin (`ToolbarFilter`). So, we will declare a new getter for this task and name it `toolbarFilterDataItems`. The value of this getter will be calculated based on the value of the `rows` getter. For this task we will need to use computed functions. These functions calculate values based on values of other getters. As you remember, we will need not only filter our grid, but display the filtered column name near the select box. Since we are going to do that in the `ToolbarFilter` plugin we will need to get the column name in the plugin. A new getter will help us do this. Let's name it `toolbarFilterColumnName`. Moreover, we will use this getter for calculating a value for `toolbarFilterDataItems`. Add a new calculated function named, for example, `filterDataItemsComputed` that accepts the `rows` and `toolbarFilterColumnName` getters and calculates items for our select box. Our code might be as follows:
```

Comment:
> For this task we will need to use computed functions.

It sounds like the functions are computed. Actually, the value of a getter is computed using the values of other getters. Can we paraphrase this somehow? For instance,  "For this task, we will use functions that we internally call 'computeds'".